### PR TITLE
BZ1990455: Added importing of image stream tags

### PR DESCRIPTION
**Pull request by snarayan-redhat**
Version(s): 4.8+

Issue: https://bugzilla.redhat.com/show_bug.cgi?id=1990455

Link to docs preview: https://55039--docspreview.netlify.app/openshift-enterprise/latest/post_installation_configuration/cluster-tasks.html#images-imagestream-import_post-install-cluster-tasks

QE review:
- [x] QE has approved this change.

SME review:
- [x] SME has approved this change

### CODE REVIEW

**Comment by flavianmissi** [comment]
Looking good @snarayan-redhat! What do you think about mentioning that for the user to configure periodic import for all cluster samples operator image stream tags they need to apply the procedure to every image stream in the namespace `openshift`?

**Comment by snarayan-redhat** [comment]
> Looking good @snarayan-redhat! What do you think about mentioning that for the user to configure periodic import for all cluster samples operator image stream tags they need to apply the procedure to every image stream in the namespace `openshift`?

Thanks for reviewing @flavianmissi ! I have tried to state the same in second step:
 _Schedule periodic import of images for **each** tag present in the image stream_
 
 If you feel this lacks emphasis, I can split this into a separate sentence. WDYT?

**Comment by flavianmissi** [comment]
Hey @snarayan-redhat. 
That sentence is good and correct 😄 
What I'm trying to point is that the customer will have to _Schedule periodic import of images for each tag present in the image stream_ for every image stream in the openshift namespace. This is what I mentioned on slack that they'll have to do multiple times, for every tag in every image stream. It's like the docs are skipping a level where first they need to find out which image streams to update (every single one on the openshift namespace), then for each of those image streams, they need to set the tag to update periodically.

In other words, periodic imports need to be enabled for each tag in every image stream in the openshift namespace.

Hope I'm not creating more confusion 😅

**Comment by snarayan-redhat** [comment]
@flavianmissi - Got that. So in short every tag of every image stream in openshift namespace. Will add that note.

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+----
+oc get imagestream <image-stream-name> -ojsonpath="{range .spec.tags[*]}Name: {.name}{'\t'}Scheduled: {.importPolicy.scheduled}{'\n'}{end}" -nopenshift
+----
++
+.Example output
+[source,terminal]
+----
+Name: 1.11	Scheduled: true
+Name: 1.12	Scheduled: true
+----
```
Comment (flavianmissi) [inline modules/images-cluster-sample-imagestream-import.adoc:67]: Just one nit: replace "Name" with "Tag", as in the below suggestion:

```suggestion
oc get imagestream <image-stream-name> -ojsonpath="{range .spec.tags[*]}Tag: {.name}{'\t'}Scheduled: {.importPolicy.scheduled}{'\n'}{end}" -nopenshift
----
+
.Example output
[source,terminal]
----
Tag: 1.11	Scheduled: true
Tag: 1.12	Scheduled: true
----
```

**Review by flavianmissi — COMMENTED** [review]
Looks great @snarayan-redhat ! left a small suggestion

**Comment by flavianmissi** [comment]
LGTM!

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+[source, terminal]
+----
+$ oc get is <image-stream-name> -o jsonpath="{range .spec.tags[*]}{.name}{'\t'}{.from.name}{'\n'}{end}"
+----
++
+For example:
++
+[source, terminal]
+----
+$ oc get is ubi8-openjdk-17 -o jsonpath="{range .spec.tags[*]}{.name}{'\t'}{.from.name}{'\n'}{end}"
```
Comment (xiuwang) [inline modules/images-cluster-sample-imagestream-import.adoc:28]: Better to add namespace `-nopenshift` in command.

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+----
++
+.Example output
+[source, terminal]
+----
+1.11	registry.access.redhat.com/ubi8/openjdk-17:1.12
```
Comment (xiuwang) [inline modules/images-cluster-sample-imagestream-import.adoc:34]: `registry.access.redhat.com/ubi8/openjdk-17:1.11`?

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+[source,terminal]
+----
+$ oc tag <repository/image> <image-stream-name:tag> --scheduled -nopenshift
+----
++
+For example:
++
+[source,terminal]
+----
+$ oc tag registry.access.redhat.com/ubi8/openjdk-17:1.12 ubi8-openjdk-17:1.11 --scheduled -nopenshift
```
Comment (xiuwang) [inline modules/images-cluster-sample-imagestream-import.adoc:49]: `registry.access.redhat.com/ubi8/openjdk-17:1.11`?

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+----
++
+This command causes {product-title} to periodically update this particular image stream tag. This period is a cluster-wide setting set to 15 minutes by default.
+
+. Verify the scheduling status of periodic import:
++
+[source,terminal]
+----
+oc get imagestream <image-stream-name> -ojsonpath="{range .spec.tags[*]}Tag: {.name}{'\t'}Scheduled: {.importPolicy.scheduled}{'\n'}{end}" -nopenshift
+----
```
Comment (xiuwang) [inline modules/images-cluster-sample-imagestream-import.adoc:60]: Add example command as above.
`oc get imagestream ubi8-openjdk-17 -ojsonpath="{range .spec.tags[*]}Tag: {.name}{'\t'}Scheduled: {.importPolicy.scheduled}{'\n'}{end}" -nopenshift`

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+[source,terminal]
+----
+$ oc tag <repositiory/image> <image-stream-name:tag>
+----
++
+For example:
++
+[source,terminal]
+----
+$ oc tag registry.access.redhat.com/ubi8/openjdk-17:1.12 ubi8-openjdk-17:1.11 -nopenshift
```
Comment (xiuwang) [inline modules/images-cluster-sample-imagestream-import.adoc:80]: `registry.access.redhat.com/ubi8/openjdk-17:1.11`

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+[source,terminal]
+----
+oc get imagestreams -nopenshift
+----
+
+. Fetch the tags for every imagestream in the previous output:
++
+[source, terminal]
+----
+$ oc get is <image-stream-name> -o jsonpath="{range .spec.tags[*]}{.name}{'\t'}{.from.name}{'\n'}{end}"
```
Comment (xiuwang) [inline modules/images-cluster-sample-imagestream-import.adoc:21]: Better to add namespace -nopenshift in command.

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+----
+Tag: 1.11	Scheduled: true
+Tag: 1.12	Scheduled: true
+----
+
+. Remove the periodic check, re-run above command but omit the `--scheduled` flag. This will reset its behavior to default.
++
+[source,terminal]
+----
+$ oc tag <repositiory/image> <image-stream-name:tag>
```
Comment (xiuwang) [inline modules/images-cluster-sample-imagestream-import.adoc:73]: Better to add namespace -nopenshift in command.

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+----
++
+.Example output
+[source, terminal]
+----
+1.11	registry.access.redhat.com/ubi8/openjdk-17:1.11
+1.12	registry.access.redhat.com/ubi8/openjdk-17:1.12
+----
+
+. Schedule periodic import of images for each tag present in the image stream by running the following command:
```
Comment (mburke5678) [inline modules/images-cluster-sample-imagestream-import.adoc:38]: s/periodic import/the periodic importing

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+[source,terminal]
+----
+oc get imagestreams -nopenshift
+----
+
+. Fetch the tags for every imagestream in the `openshift` namespace by running the following command:
++
+[source, terminal]
+----
+$ oc get is <image-stream-name> -o jsonpath="{range .spec.tags[*]}{.name}{'\t'}{.from.name}{'\n'}{end}" -nopenshift
```
Comment (mburke5678) [inline modules/images-cluster-sample-imagestream-import.adoc:21]: s/-nopenshift/-n openshift

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+----
+1.11	registry.access.redhat.com/ubi8/openjdk-17:1.11
+1.12	registry.access.redhat.com/ubi8/openjdk-17:1.12
+----
+
+. Schedule periodic import of images for each tag present in the image stream by running the following command:
++
+[source,terminal]
+----
+$ oc tag <repository/image> <image-stream-name:tag> --scheduled -nopenshift
```
Comment (mburke5678) [inline modules/images-cluster-sample-imagestream-import.adoc:42]: s/-nopenshift/-n openshift

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+----
++
+For example:
++
+[source,terminal]
+----
+$ oc tag registry.access.redhat.com/ubi8/openjdk-17:1.11 ubi8-openjdk-17:1.11 --scheduled -nopenshift
+$ oc tag registry.access.redhat.com/ubi8/openjdk-17:1.12 ubi8-openjdk-17:1.12 --scheduled -nopenshift
```
Comment (mburke5678) [inline modules/images-cluster-sample-imagestream-import.adoc:50]: s/-nopenshift/-n openshift

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+----
++
+This command causes {product-title} to periodically update this particular image stream tag. This period is a cluster-wide setting set to 15 minutes by default.
+
+. Verify the scheduling status of periodic import by running the following command:
```
Comment (mburke5678) [inline modules/images-cluster-sample-imagestream-import.adoc:55]: s/periodic/the periodic

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
++
+[source,terminal]
+----
+oc get imagestream <image-stream-name> -o jsonpath="{range .spec.tags[*]}Tag: {.name}{'\t'}Scheduled: {.importPolicy.scheduled}{'\n'}{end}" -nopenshift
+----
```
Comment (mburke5678) [inline modules/images-cluster-sample-imagestream-import.adoc:60]: s/-nopenshift/-n openshift

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
++
+.Example output
+[source,terminal]
+----
+Tag: 1.11	Scheduled: true
+Tag: 1.12	Scheduled: true
+----
+
+. Remove the periodic import by running the following command:
```
Comment (mburke5678) [inline modules/images-cluster-sample-imagestream-import.adoc:76]: In the modular docs theory, this step should be in it's own module, as it is a separate task.

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
++
+[source,terminal]
+----
+$ oc tag <repository/image> <image-stream-name:tag> -nopenshift
```
Comment (mburke5678) [inline modules/images-cluster-sample-imagestream-import.adoc:80]: s/-nopenshift/-n openshift

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+----
++
+For example:
++
+[source,terminal]
+----
+$ oc tag registry.access.redhat.com/ubi8/openjdk-17:1.11 ubi8-openjdk-17:1.11 -nopenshift
+$ oc tag registry.access.redhat.com/ubi8/openjdk-17:1.12 ubi8-openjdk-17:1.12 -nopenshift
```
Comment (mburke5678) [inline modules/images-cluster-sample-imagestream-import.adoc:88]: s/-nopenshift/-n openshift

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 18]
```diff
@@ -0,0 +1,89 @@
+// Module included in the following assemblies:
+// * openshift_images/cluster-tasks.adoc
+
+:_content-type: PROCEDURE
+[id="images-cluster-sample-imagestream-import_{context}"]
+= Configuring periodic importing of Cluster Sample Operator image stream tags
+
```
Comment (mburke5678) [inline modules/images-cluster-sample-imagestream-import.adoc:7]: Should we put an introductory paragraph to tell folks what they are doing and why?

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+.Procedure
+
+. Fetch all the imagestreams in the `openshift` namespace by running the following command:
++
+[source,terminal]
+----
+oc get imagestreams -nopenshift
+----
+
```
Comment (mburke5678) [inline modules/images-cluster-sample-imagestream-import.adoc:16]: Do we want example output for this command?

File: post_installation_configuration/cluster-tasks.adoc [hunk 20]
```diff
@@ -706,3 +706,5 @@ include::modules/installation-images-samples-disconnected-mirroring-assist.adoc[
 include::modules/installation-restricted-network-samples.adoc[leveloffset=+2]
 
 include::modules/installation-preparing-restricted-cluster-to-gather-support-data.adoc[leveloffset=+2]
+
+include::modules/images-cluster-sample-imagestream-import.adoc[leveloffset=+1]
```
Comment (mburke5678) [inline post_installation_configuration/cluster-tasks.adoc:710]: Should this be `[leveloffset=+2]` to match the Cluster Sample Operator modules before it?

**Comment by mburke5678** [comment]
@snarayan-redhat A few nits. Otherwise LGTM

File: modules/images-cluster-sample-imagestream-import.adoc (hunk 9 above)
Comment (snarayan-redhat) [reply modules/images-cluster-sample-imagestream-import.adoc:38]: Ack

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+[id="images-cluster-sample-imagestream-import_{context}"]
+= Configuring periodic importing of Cluster Sample Operator image stream tags
+
+.Procedure
+
+. Fetch all the imagestreams in the `openshift` namespace by running the following command:
++
+[source,terminal]
+----
+oc get imagestreams -nopenshift
```
[the remark this answers is not in the document]
Comment (snarayan-redhat) [reply modules/images-cluster-sample-imagestream-import.adoc:14]: `-nopenshift` is same as `-n openshift`

File: modules/images-cluster-sample-imagestream-import.adoc (hunk 10 above)
Comment (snarayan-redhat) [reply modules/images-cluster-sample-imagestream-import.adoc:21]: `-nopenshift` is same as `-n openshift`

File: modules/images-cluster-sample-imagestream-import.adoc [hunk 22]
```diff
@@ -0,0 +1,89 @@
...
+[source, terminal]
+----
+$ oc get is <image-stream-name> -o jsonpath="{range .spec.tags[*]}{.name}{'\t'}{.from.name}{'\n'}{end}" -nopenshift
+----
++
+For example:
++
+[source, terminal]
+----
+$ oc get is ubi8-openjdk-17 -o jsonpath="{range .spec.tags[*]}{.name}{'\t'}{.from.name}{'\n'}{end}" -nopenshift
```
[the remark this answers is not in the document]
Comment (snarayan-redhat) [reply modules/images-cluster-sample-imagestream-import.adoc:28]: `-nopenshift` is same as `-n openshift`

File: modules/images-cluster-sample-imagestream-import.adoc (hunk 11 above)
Comment (snarayan-redhat) [reply modules/images-cluster-sample-imagestream-import.adoc:42]: `-nopenshift` is same as `-n openshift`

File: modules/images-cluster-sample-imagestream-import.adoc (hunk 12 above)
Comment (snarayan-redhat) [reply modules/images-cluster-sample-imagestream-import.adoc:50]: `-nopenshift` is same as `-n openshift`

File: modules/images-cluster-sample-imagestream-import.adoc (hunk 13 above)
Comment (snarayan-redhat) [reply modules/images-cluster-sample-imagestream-import.adoc:55]: Ack

File: modules/images-cluster-sample-imagestream-import.adoc (hunk 14 above)
Comment (snarayan-redhat) [reply modules/images-cluster-sample-imagestream-import.adoc:60]: `-nopenshift` is same as `-n openshift`

File: modules/images-cluster-sample-imagestream-import.adoc (hunk 15 above)
Comment (snarayan-redhat) [reply modules/images-cluster-sample-imagestream-import.adoc:76]: Thanks for catching this.

File: modules/images-cluster-sample-imagestream-import.adoc (hunk 16 above)
Comment (snarayan-redhat) [reply modules/images-cluster-sample-imagestream-import.adoc:80]: `-nopenshift` is same as `-n openshift`

File: modules/images-cluster-sample-imagestream-import.adoc (hunk 18 above)
Comment (snarayan-redhat) [reply modules/images-cluster-sample-imagestream-import.adoc:7]: That would be good to have.
